### DISCUSSION TITLE
How to plot directly in color leaving the axis in the default color

### DISCUSSION THREAD

**Discussion by brandy125** [gnudatalanguage/gdl#1895] [Q&A]
Is it possible to plot directly in the color red leaving the coordinate system as it is in default?

I always need to plot twice like:

```
GDL> plot,findgen(100)
GDL>oplot,findgen(100),color=255
```

If it does not exist, shouldn't we implement it ?

**Comment by alaingdl** [comment]
To my knowledge it is not directly possible in one internal command IDL (we cannot separate the colors of lines)
But if is quite easy to prepare a superset procedure to manage that. E.G. this is what CGplot does in in Coyote library
```
GDL> #cgplot
Compiled PROCEDURE : CGPLOT,[2 Args],_REF_EXTRA,YTITLE,YSTYLE,YRANGE,YMARGIN,XTITLE,XSTYLE,XRANGE,XMARGIN,WINDOW,TRADITIONAL,TITLE,SUBTITLE,SYMSIZE,SYMCOLOR,PSYM,POSITION,OVERPLOT,OUTPUT,OUTFILENAME,OPLOTS,NOERASE,NODATA,MAP_OBJECT,LEGENDS,LAYOUT,LABEL,ISOTROPIC,FONT,ERR_YLOW,ERR_YHIGH,ERR_XLOW,ERR_XHIGH,ERR_WIDTH,ERR_THICK,ERR_COLOR,ERR_CLIP,COLOR,CHARSIZE,BACKGROUND,AXESCOLOR,AXISCOLOR,ASPECT,ADDCMD

GDL> cgPlot, Findgen(11), Color=cgColor("blue"), AXESCOLOR=cgColor("red")

```

Unfortunately the original site seems to be discontinued but copies around ...
https://www.higp.hawaii.edu/~harold/GSE_ADA_doc/coyote/cgcolor.html#cgColor

**Comment by brandy125** [comment]
Thanks for answering!
Thought it would be easy to make it happen.
Any chance to introduce a new keyword like pcolor=255 that could provide this feature?

**Comment by GillesDuvert** [comment]
IDL (and GDL) provide the NODATA keyword just for that: writing axis, titles etc in one color and leaving the command OPLOT to plot the data in another color.
In the (not yet implemented in GDL) PLOT function (equivalent of what coyote did , and IMHO  quite certainly inspired by it) the function permits such things directly.
Reusing or adpating the coyote procedures to mimic the IDL graphic functions (as opposed to todays procedures) should be easy, if not quick, for a reasonable IDL expert, no C/C+ skills required...)

**Comment by brandy125** [comment]
Ok, thanks!
